### PR TITLE
feat(DataSpreadsheet): add initial component

### PR DESCRIPTION
**Pull request by matthewgallo**
Contributes to #1474 

This adds the initial DataSpreadsheet component. The first version is a read only without any of the interactions, with the incorporation of `react-table` similar to the WKC data table that will be contributed to our library (allowing us to more easily virtualize the rows). I added two stories, a basic and another with 1000 rows.

image

#### What did you change?
```
packages/cloud-cognitive/package.json
packages/cloud-cognitive/src/components/DataSpreadsheet/DataSpreadsheet.js
packages/cloud-cognitive/src/components/DataSpreadsheet/DataSpreadsheet.mdx
packages/cloud-cognitive/src/components/DataSpreadsheet/DataSpreadsheet.stories.js
packages/cloud-cognitive/src/components/DataSpreadsheet/DataSpreadsheet.test.js
packages/cloud-cognitive/src/components/DataSpreadsheet/_data-spreadsheet.scss
packages/cloud-cognitive/src/components/DataSpreadsheet/_index.scss
packages/cloud-cognitive/src/components/DataSpreadsheet/_storybook-styles.scss
packages/cloud-cognitive/src/components/DataSpreadsheet/generateData.js
packages/cloud-cognitive/src/components/DataSpreadsheet/index.js
packages/cloud-cognitive/src/components/_index.scss
packages/cloud-cognitive/src/components/index.js
packages/cloud-cognitive/src/global/js/package-settings.js
packages/cloud-cognitive/src/global/js/utils/getScrollbarWidth.js
packages/core/story-structure.js
yarn.lock
```
#### How did you test and verify your work?
Storybook and ran tests

### CODE REVIEW

File: packages/cloud-cognitive/src/components/DataSpreadsheet/_storybook-styles.scss [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//
+
+@import './index';
+
+// TODO: add any additional styles used by DataSpreadsheet.stories.js
+
+// Uncomment next line (which must appear last) to test in storybook
+// that the SCSS styles for this component are sufficiently specific
+// to override Carbon whichever order the styles get loaded in.
+//@import 'carbon-components/css/carbon-components.min';
```
Comment (dcwarwick) [inline packages/cloud-cognitive/src/components/DataSpreadsheet/_storybook-styles.scss:15]: This was an error, and a bit of obsolete commentary, in the cli templates, sorry, which I've just fixed in another PR. We should not import ./index here as it duplicates styles in storybook.

```suggestion
// TODO: add any additional styles used by DataSpreadsheet.stories.js

```

**Review by dcwarwick — CHANGES_REQUESTED** [review]
looks good, just one suggestion.

We also need to be careful about adding dependencies that will add to the weight of our package to users who don't need the components that require them. This is just a comment, not a suggestion to do anything differently, but something for us to think about.

File: packages/cloud-cognitive/src/components/DataSpreadsheet/_storybook-styles.scss (hunk 1 above)
Comment (matthewgallo) [reply packages/cloud-cognitive/src/components/DataSpreadsheet/_storybook-styles.scss]: Removed [here](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/1606/commits/005fa39e3e8fd0862820232c0b9d121e0e612a1e)

**Comment by matthewgallo** [comment]
Yes I absolute agree with @dcwarwick about us being cautious of the dependencies we add. @davidmenendez these dependencies are required by the new data table that the WKC team is contributing, so we can double check with them.

**Comment by dcwarwick** [comment]
> have you verified that we're ok to use `react-table`

@matthewgallo I've remembered that we [have a checklist](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/main/docs/reviews/PR_REVIEW_GUIDELINES.md) for adding new production dependencies. We'd need to do this for both `react-table` and `react-window`, if we will definitely need both of those.

**Comment by matthewgallo** [comment]
## Review of dependencies

- [X] Verify the dependency has previously been [pedigree reviewed](https://pedigree-service.wdc1a.cirrus.ibm.com/)
Screen Shot 2022-02-15 at 8 04 24 PM
Screen Shot 2022-02-15 at 7 51 12 PM

- [X] Verify the dependency [doesn't contain any vulnerabilities](https://snyk.io/vuln) (**Ran `yarn audit` and no vulnerabilities were produced from either of the two packages, however we do have some vulnerabilities coming from `stylelint-config-carbon` and `jest-config-ibm-cloud-cognitive`)
- [X] Verify the dependency [bundle size](https://bundlephobia.com/) is acceptable, bundle size comparison of the two packages as compared to `carbon-components` and `carbon-components-react`
Screen Shot 2022-02-10 at 9 50 36 AM
Screen Shot 2022-02-10 at 9 50 21 AM
Screen Shot 2022-02-15 at 8 01 59 PM
Screen Shot 2022-02-15 at 8 01 41 PM
 
- [X] Verify the dependency is [actively maintained](https://www.npmtrends.com/), screenshots from npmtrends showing uptrend usage over the past 2 years
Screen Shot 2022-02-15 at 8 03 22 PM
Screen Shot 2022-02-15 at 8 02 57 PM

- [X] Verify license conditions are met e.g. forwarding of copyright if necessary (both packages use MIT licensing).

**Review by dcwarwick — APPROVED** [review]
LGTM. Thanks Matt.

**Pull request merged**